ast/src/main/java/com/graphicsfuzz/common/util/UpgradeShadingLanguageVersion.java

```diff
@@ -92,6 +173,73 @@ public void visitVariableIdentifierExpr(VariableIdentifierExpr variableIdentifie
     }
   }
 
+  @Override
+  public void visitVariablesDeclaration(VariablesDeclaration variablesDeclaration) {
+    final Type baseType = variablesDeclaration.getBaseType();
+    if (baseType.hasQualifier(TypeQualifier.VARYING)) {
+      ((QualifiedType) baseType).replaceQualifier(TypeQualifier.VARYING,
+          TypeQualifier.SHADER_INPUT);
+    }
+    visit(baseType);
+    for (VariableDeclInfo vdi : variablesDeclaration.getDeclInfos()) {
+      if (vdi.hasArrayInfo()) {
+        visit(vdi.getArrayInfo().getSizeExpr());
+      } else if (baseType instanceof ArrayType) {
+        visit(((ArrayType) baseType).getArrayInfo().getSizeExpr());
+      }
+      if (vdi.hasInitializer()) {
+        visit(vdi.getInitializer());
+        // If not constant and has initializer, move the initializer to main.
+        if (!baseType.hasQualifier(TypeQualifier.CONST)) {
+          globals.add(new ExprStmt(new BinaryExpr(new VariableIdentifierExpr(vdi.getName()),
+              vdi.getInitializer().getExpr(), BinOp.ASSIGN)));
```

Comment:
I would slightly prefer us to have a "removeInitizlizer()" method on VariableIdentifierExpr, rather than explicitly setting the initializer to 'null'; that method would do the nullifying.  Would you be able to add such a method (with brief JavaDoc) and invoke it from here?

Reply:
Done

Reply:
(Also added other javadoc comments to the file)